src/Agent/NewRelic/Agent/Core/Configuration/ReportedConfiguration.cs

```diff
@@ -1,114 +1,607 @@
+        public int ErrorCollectorMaxEventSamplesStored => _configuration.ErrorCollectorMaxEventSamplesStored;
 
-        [JsonProperty("error_collector.ignore_messages")]
-        public IDictionary<string, IEnumerable<string>> ErrorCollectorIgnoreMessages { get; set; }
+        [JsonProperty("error_collector.harvest_cycle")]
+        public TimeSpan ErrorEventsHarvestCycle => _configuration.ErrorEventsHarvestCycle;
+
+        [JsonProperty("error_collector.max_per_period")]
+        public uint ErrorsMaximumPerPeriod => _configuration.ErrorsMaximumPerPeriod;
+
+        [JsonProperty("error_collector.expected_status_codes")]
+        public IEnumerable<MatchRule> ExpectedStatusCodes => _configuration.ExpectedStatusCodes;
 
         [JsonProperty("error_collector.expected_classes")]
-        public IEnumerable<string> ErrorCollectorExpectedClasses { get; set; }
+        public IEnumerable<string> ExpectedErrorClassesForAgentSettings => _configuration.ExpectedErrorClassesForAgentSettings;
 
         [JsonProperty("error_collector.expected_messages")]
-        public IDictionary<string, IEnumerable<string>> ErrorCollectorExpectedMessages { get; set; }
+        public IDictionary<string, IEnumerable<string>> ExpectedErrorMessagesForAgentSettings => _configuration.ExpectedErrorMessagesForAgentSettings;
 
-        [JsonProperty("error_collector.expected_status_codes")]
-        public IEnumerable<string> ErrorCollectorExpectedStatusCodes { get; set; }
+        // TODO: Less snark
+        [JsonProperty("error_collector.expected_status_codes_for_agent_settings_trolololol")]
+        public IEnumerable<string> ExpectedErrorStatusCodesForAgentSettings => _configuration.ExpectedErrorStatusCodesForAgentSettings;
 
-        [JsonProperty("transaction_tracer.stack_trace_threshold")]
-        public double TransactionTracerStackThreshold { get; set; }
+        [JsonProperty("error_collector.expected_errors_config")]
+        public IDictionary<string, IEnumerable<string>> ExpectedErrorsConfiguration => _configuration.ExpectedErrorsConfiguration;
+
+        [JsonProperty("error_collector.ignore_errors_config")]
+        public IDictionary<string, IEnumerable<string>> IgnoreErrorsConfiguration => _configuration.IgnoreErrorsConfiguration;
+
+        [JsonProperty("error_collector.ignore_classes")]
+        public IEnumerable<string> IgnoreErrorClassesForAgentSettings => _configuration.IgnoreErrorClassesForAgentSettings;
+
+        [JsonProperty("error_collector.ignore_messages")]
+        public IDictionary<string, IEnumerable<string>> IgnoreErrorMessagesForAgentSettings => _configuration.IgnoreErrorMessagesForAgentSettings;
+
+        [JsonProperty("agent.request_headers_map")]
+        public Dictionary<string, string> RequestHeadersMap => _configuration.RequestHeadersMap;
+
+        // Use old name
+        //[JsonProperty("agent.encoding_key")]
+        [JsonProperty("encoding_key")]
+        public string EncodingKey => _configuration.EncodingKey;
```

Comment:
What is this encoding key, and is it something we're concerned about leaking?

Reply:
It looks like this has to do with CAT functionality... and it was sent up by the previous 'ReportedConfiguration', so apparently it's not a security concern :)